silhouette/app/com/mohiva/play/silhouette/impl/providers/state/CsrfStateItemHandler.scala

```diff
@@ -0,0 +1,167 @@
+   *
+   * @param item The state item to serialize.
+   * @return The serialized state item.
+   */
+  override def serialize(item: Item): ItemStructure = ItemStructure(ID, Json.toJson(item))
+
+  /**
+   * Unserializes the state item.
+   *
+   * @param item    The state item to unserialize.
+   * @param request The request instance to get additional data to validate against.
+   * @param ec      The execution context to handle the asynchronous operations.
+   * @tparam B The type of the request body.
+   * @return The unserialized state item.
+   */
+  override def unserialize[B](item: ItemStructure)(implicit request: ExtractableRequest[B], ec: ExecutionContext): Future[Item] = {
+    Future.fromTry(Try(item.data.as[Item]))
+  }
+
+  /**
+   * Publishes the Csrf State to the client.
+   *
+   * @param item    The item to publish.
+   * @param result  The result to send to the client.
+   * @param request The current request.
+   * @tparam B The type of the request body.
+   * @return The result to send to the client.
+   */
+  override def publish[B](item: Item, result: Result)(implicit request: ExtractableRequest[B]): Result = {
+    result.withCookies(Cookie(
+      name = settings.cookieName,
+      value = cookieSigner.sign(item.value),
+      maxAge = Some(settings.expirationTime.toSeconds.toInt),
+      path = settings.cookiePath,
+      domain = settings.cookieDomain,
+      secure = settings.secureCookie,
+      httpOnly = settings.httpOnlyCookie))
+  }
+
+  /**
+   * Gets the Csrf State from the cookie.
+   *
+   * @param request The request header.
+   * @return The OAuth2 state on success, otherwise a failure.
+   */
+  private def clientState(implicit request: RequestHeader): Try[Item] = {
+    request.cookies.get(settings.cookieName) match {
+      case Some(cookie) => cookieSigner.extract(cookie.value).map(token => CsrfState(token))
+      case None         => Failure(new OAuth2StateException(ClientStateDoesNotExists.format(settings.cookieName)))
+    }
+  }
+}
+
+/**
+ * The companion object.
+ */
+object CsrfStateItemHandler {
+  val ID = "csrf-state"
+  /**
+   * The error messages.
+   */
+  val ClientStateDoesNotExists = "[Silhouette][CookieState] State cookie doesn't exists for name: %s"
+  val ProviderStateDoesNotExists = "[Silhouette][CookieState] Couldn't find state in request for param: %s"
+  val StateIsNotEqual = "[Silhouette][CookieState] State isn't equal"
+  val StateIsExpired = "[Silhouette][CookieState] State is expired"
+  val InvalidJson = "[Silhouette][CookieState] Cannot parse invalid Json: %s"
+  val InvalidStateFormat = "[Silhouette][CookieState] Cannot build OAuth2State because of invalid Json format: %s"
+  val InvalidCookieSignature = "[Silhouette][CookieState] Invalid cookie signature"
+
+  /**
+   * Json Format for the Csrf State
+   */
+  implicit val csrfFormat: Format[CsrfState] = Json.format[CsrfState]
+}
+
+/**
+ * The settings for the Csrf State.
+ *
+ * @param cookieName The cookie name.
```

Comment:
We should be consistent with the alignment of the multi param description. For new code we use a one line alignment. Please see the above doc comments.